data/index.js

```diff
@@ -134,15 +134,17 @@ function updateCourseGrade(studentId, courseId, grade) {
 function calculateAndUpdateCourseGrade(studentId, courseId) {
 	return getCourse(courseId).then((course) => {
 		let grade = 0;
+		let graded_assignments = 0;
 		return module.exports.getAssignmentsForCourse(courseId).then((assignments) => {
 			assignments.forEach((assignment) => {
 				assignment.submissions.forEach((submission) => {
-					if (submission.studentId === studentId) {
+					if ((submission.studentId === studentId) && (grade !== NaN)) {
```

Comment:
the !== NaN does work
you have to use the function isNaN
in javascript NaN does not equal NaN